src/Utilities/struct_of_constraints.jl

```diff
@@ -223,62 +212,61 @@ If `types` is vector of `SymbolFun` (resp. `SymbolSet`) then the constraints
 of that function (resp. set) type are stored in the corresponding field.
 """
 function struct_of_constraint_code(struct_name, types, field_types = nothing)
-    esc_struct_name = struct_name
     T = esc(:T)
-    typed_struct = :($(esc_struct_name){$T})
+    typed_struct = :($(struct_name){$T})
     type_parametrized = field_types === nothing
     if type_parametrized
         field_types = [Symbol("C$i") for i in eachindex(types)]
```

Comment:
@blegat ideally we could rewrite the macros to avoid all these extra type parameters, but I struggled to get everything working.

Reply:
This is already an option, that's what `@model` does for the function layer, but not for the set layer.
Otherwise, the type of `MOIU.Model` is gigantic :D